### ISSUE TITLE
plone.rest enters an infinite redirection loop if path is not found

### ISSUE THREAD

**Issue by ericof** [plone/plone.rest#133]
If a user adds a redirect from a parent to a child:

```python
storage = queryUtility(IRedirectionStorage)
storage.add("/plone/folder-new", "/plone/folder-new/archive")
```
And there is a request that should be a 404, it becomes an infinite loop:
```python
requests.get(
    self.portal_url + "/folder-new/sub_folder/not-found",
    headers={"Accept": "application/json"},
    auth=(SITE_OWNER_NAME, SITE_OWNER_PASSWORD),
)
```

**Comment by ericof** [comment]
This is related to [plone.restapi #1357](https://github.com/plone/plone.restapi/pull/1357)

**Comment by ericof** [comment]
Closed with #134
